core/templates/dev/head/pages/exploration_player/PredictionAlgorithmRegistryService.js

```diff
@@ -0,0 +1,39 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for mapping algorithmId to PredictionAlgorithmService.
+ */
+
+ oppia.factory('PredictionAlgorithmRegistryService', [
+   'PredictionAlgorithmSampleService',
+   function(PredictionAlgorithmSampleService) {
+     // This mapping needs to be updated whenever a new prediction service needs
+     // to be added for classification.
+     var algorithmIdPredictionServiceMapping = {};
+
+     return {
+       getPredictionService: function(algorithmId) {
+         if (algorithmIdPredictionServiceMapping.hasOwnProperty(algorithmId)) {
+           return algorithmIdPredictionServiceMapping[algorithmId];
+         } else {
+           return null;
+         }
+       },
+       setMapping: function(newAlgorithmIdPredictionServiceMapping) {
```

Comment:
@prasanna08. Yes.The mapping variable inside the registry service has to be empty and not use testing sample services when we push to develop. So, we use this setMapping method while writing Karma tests.